From d01d272ca2f78d9da50ab7835e825c5ddce35d26 Mon Sep 17 00:00:00 2001
From: "Shankarmurthy, Akshay" <akshay.s@ti.com>
Date: Mon, 26 Dec 2011 11:14:48 +0530
Subject: [PATCH 0766/1092] pwm:ehrpwm - Fix duty cycle inversion eHRPWM wave.

Patch fixes the ON and OFF period inversion in PWM wave.
Action qualifier register settings causes switching PWM wave to OFF
period when counter reaches zero and to ON period on counter reaches
compare value. Because of this inversion in duty cycle was observed.

Signed-off-by: Shankarmurthy, Akshay <akshay.s@ti.com>
Signed-off-by: Philip, Avinash <avinashphilip@ti.com>
---
 drivers/pwm/ehrpwm.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/pwm/ehrpwm.c b/drivers/pwm/ehrpwm.c
index 9b48d75..c0d3456 100644
--- a/drivers/pwm/ehrpwm.c
+++ b/drivers/pwm/ehrpwm.c
@@ -978,9 +978,10 @@ static int ehrpwm_pwm_set_pol(struct pwm_device *p)
 		ctreqcmp = 8;
 	}
 
-	val = ((p->active_high ? ACTCTL_CTREQCMP_LOW : ACTCTL_CTREQCMP_HIGH)
-		 << ctreqcmp) | (p->active_high ? ACTCTL_CTREQZRO_HIGH :
-			ACTCTL_CTREQZRO_LOW);
+
+	val = ((p->active_high ? ACTCTL_CTREQCMP_HIGH : ACTCTL_CTREQCMP_LOW)
+		 << ctreqcmp) | (p->active_high ? ACTCTL_CTREQZRO_LOW :
+			ACTCTL_CTREQZRO_HIGH);
 	ehrpwm_write(ehrpwm, act_ctrl_reg, val);
 
 	return 0;
-- 
1.7.11.2

